Separate API section (#3)
mercari-build/mercari-build-training-2023: document/04-database.en.md

* Install SQlite, and make a database file called `mercari.sqlite3`.  
* Open this file and make a table called `items`. 
* Define the items table as follows and save the schema into `db/items.db`.
  * id: int (Identifier unique for each item)
  * name: string (Name of the item)
  * category: string (Category of the item)
  * image_name: string (Filename of the imeage)

Change the endpoints `GET /items` and `POST /items` such that items are saved into the database and can be returned based on GET request. Add `mercari.sqlite3` to your `.gitignore` such that it is not commited. `items.db` shoud be commited. 

**:beginner: Points**

* What are the advantages of saving into a database such as SQLite instead of saving into a single JSON file?

## 2. Search for an item

Make an endpoint to return a list of items that include a specified keyword called `GET /search`.

```shell
# Request a list of items containing string "jacket"
$ curl -X GET 'http://127.0.0.1:9000/search?keyword=jacket'
# Expected response for a list of items with name containing "jacket"
{"items": [{"name": "jacket", "category": "fashion"}, ...]}
```

## 8. Move the category information to a separate table

> Modify the database as follows. This allows changes in the category names and you will not have to change all categories in the items table when they do. Since `GET items` should still return the name of the category, join these two tables when returning responses.

Reviewer: ```suggestion
Modify the database as follows. This makes it possible to change the category names without modifying the all categories of items in the items table.
Since `GET items` should return the category name as before, **join** these two tables when returning responses.
```
